src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpTraceExporter.cs

```diff
@@ -63,7 +64,20 @@ public OtlpTraceExporter(OtlpExporterOptions options)
         internal OtlpTraceExporter(OtlpExporterOptions options, OtlpCollector.TraceService.ITraceServiceClient traceServiceClient = null)
         {
             this.options = options ?? throw new ArgumentNullException(nameof(options));
-            this.headers = options.Headers ?? throw new ArgumentException("Headers were not provided on options.", nameof(options));
+            this.headers = GetMetadataFromHeaders(options.Headers);
+
+            if (options.TimeOut.HasValue)
+            {
+                if (options.TimeOut.Value <= 0)
+                {
+                    throw new ArgumentException("TimeOut vaue provided is not a positive numbe", nameof(options.TimeOut));
+                }
+                else
+                {
+                    this.deadline = DateTime.UtcNow.AddMilliseconds(options.TimeOut.Value);
```

Comment:
Don't we need to calculate this in Export? If we do in ctor we'll blow by it pretty fast :smile:

Reply:
Moved the logic to determine deadline to Export method.